### PR TITLE
 todolist week-3

### PR DESCRIPTION
**Pull request by basafilm**

### CODE REVIEW

**Comment by abdualrhman** [comment]
Good job @basafilm!

I left a couple of comments for you in the code

**Comment by basafilm** [comment]
> Good job @basafilm!
> 
> I left a couple of comments for you in the code

Hi 
I can't find the comments . You may missed !

**Comment by abdualrhman** [comment]
> > Good job @basafilm!
> > I left a couple of comments for you in the code
> 
> Hi
> I can't find the comments . You may missed !

Oh, I may have not sumbitted the comments.

File: react/week3/todolist/src/App.js [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+  edit(id , e ){   
+    const todosIndex= this.state.todos.findIndex(todo => todo.id ===id)
+    const todo = Object.assign({},this.state.todos[todosIndex]);
+    todo.description =this.state.description;
+    todo.deadline =this.state.deadline;
+    const todos = Object.assign([], this.state.todos);
+    todos[todosIndex] = todo;
+    this.setState({
+      todos: todos
+    })
```
Comment (abdualrhman) [inline react/week3/todolist/src/App.js:87]: Another way determine which item to update is by comparing the updated todo id with corresponded todo id in the state, fx: 
```suggestion
 this.setState(prevState => {
            const newList = prevState.todos.map(todo => {
                if (id === todo.id) {
                    return {
                        ...todo,
                        description: prevState.description,
                        deadline: prevState.deadline,
                    };
                }
                return todo;
            });
            return { todos: newList };
        });
```
This way you don't have to worry about indexing.

**Comment by basafilm** [comment]
> > > Good job @basafilm!
> > > I left a couple of comments for you in the code
> > 
> > 
> > Hi
> > I can't find the comments . You may missed !
> 
> Oh, I may have not sumbitted the comments.

Thank for your feedback !

**Pull request merged**